spp_farmer_registry_base/models/base_import.py

```diff
@@ -15,7 +15,14 @@ def execute_import(self, fields, columns, options, dryrun=False):
             if "is_group" in fields:
                 with_is_group = True
             if "default_is_group" in self.env.context:
-                with_is_group = True
+                if self.env.context["default_is_group"]:
+                    with_is_group = True
+            if "default_kind" in self.env.context:
+                if self.env.context["default_kind"] == self.env.ref("spp_farmer_registry_base.kind_farm").id:
+                    with_is_group = True
+                else:
+                    with_is_group = False
+
             if with_is_group and not ("farmer_given_name" in fields and "farmer_family_name" in fields):
                 raise ValidationError(_("farmer_given_name and farmer_family_name must be present in the excel file."))
```

Comment:
@emjay0921 I think you can simplify the whole function to this. but test it first and modify if it will not cover some case or scenario.

```suggestion
    def execute_import(self, fields, columns, options, dryrun=False):
        if self.res_model == "res.partner":
            with_is_group = "is_group" in fields

            if self.env.context.get("default_is_group"):
                with_is_group = True
            elif "default_kind" in self.env.context:
                with_is_group = self.env.context["default_kind"] == self.env.ref("spp_farmer_registry_base.kind_farm").id

            required_fields = {"farmer_given_name", "farmer_family_name"}
            if with_is_group and not required_fields.issubset(fields):
                raise ValidationError(_("farmer_given_name and farmer_family_name must be present in the excel file."))

        return super().execute_import(fields, columns, options, dryrun=dryrun)
```